### PR TITLE
doc: releases: Add GPIO and I2S release notes for v3.1

### LINKED ISSUE #46215: Release notes for v3.1.0: GPIO
As maintainer, please send a pull request filling in the "GPIO" section(s) in doc/releases/release-notes-3.1.rst as soon as possible.

### LINKED ISSUE #46217: Release notes for v3.1.0: I2S
As maintainer, please send a pull request filling in the "I2S" section(s) in doc/releases/release-notes-3.1.rst as soon as possible.

### PR DESCRIPTION
**Pull request by anangl**
Add release notes related to GPIO and I2S drivers.

Closes #46215.
Closes #46217.

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]
Thank you, LGTM!

File: doc/releases/release-notes-3.1.rst [hunk 1]
```diff
@@ -123,6 +123,12 @@ Deprecated in this release
 * The TinyCBOR module has been deprecated in favor of the new zcbor CBOR
   library, included with Zephyr in this release.
 
+* GPIO
+
+  * Deprecated the ``GPIO_INT_DEBOUNCE`` flag and the ``GPIO_DS_*`` and
+    ``GPIO_VOLTAGE_*`` groups of flags. Controller/SoC specific flags
+    should be now used instead.
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.1.rst:130]:
```suggestion
    should now be used instead.
```

File: doc/releases/release-notes-3.1.rst [hunk 2]
```diff
@@ -459,6 +465,13 @@ Drivers and Sensors
 
 * GPIO
 
+  * Refactored GPIO flags. Upper 8 bits of ``gpio_dt_flags_t`` is now reserved
+    for controller/SoC specific flags and certain hardware-specific flags
+    defined as common so far (IO voltage level, drive strength, debounce filter)
+    were replaced with ones defined in this controller/SoC specific space.
+  * Added Xilinx PS MIO/EMIO GPIO controller driver.
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.1.rst:472]: I'm pretty sure this is just MIO, not EMIO.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.1.rst (hunk 2 above)
Comment (anangl) [reply doc/releases/release-notes-3.1.rst:472]: I took this information from https://github.com/zephyrproject-rtos/zephyr/commit/ca3390524815520c78e7bdc60990385e9a332ad2. It mentions both MIO and EMIO everywhere.

Comment (henrikbrixandersen) [reply doc/releases/release-notes-3.1.rst:472]: Got it.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**